src/descriptor/satisfied_contraints.rs

```diff
@@ -0,0 +1,1482 @@
+                                None
+                            },
+                            StackElement::Push(sigser) => {
+                                let sig = verify_sersig(
+                                    secp,
+                                    sighash,
+                                    &pk.to_public_key(),
+                                    &sigser);
+                                match sig {
+                                    Ok(sig) => {
+                                        self.push(StackElement::Satisfied);
+                                        Some(Ok(SatisfiedConstraint::PublicKeyHasH {
+                                            keyhash: pkh,
+                                            key: pk,
+                                            sig
+                                        }))
+                                    }
+                                    Err(e) => {
+                                        return Some(Err(e))
+                                    }
+                                }
+                            }
+                            StackElement::Satisfied =>
+                                Some(Err(Error::CouldnotEvaluate(
+                                    "PK evaluation error: Incorrect Signature".to_owned())))
+                        }
+                    } else {
+                        Some(Err(Error::CouldnotEvaluate(
+                            "PK evaluation error: Signature not present".to_owned())))
+                    }
+                }
+                Err(..) => Some(Err(
+                    Error::CouldnotEvaluate("Could not parse pubkey".to_owned())))
+            }
+        } else {
+            Some(Err(Error::CouldnotEvaluate("Pkh evaluation failed".to_owned())))
+        }
+    }
+
+    /// Helper function to evaluate a After and AfterF Node. Takes no arugment from stack
+    /// `n CHECKSEQUENCEVERIFY 0NOTEQUAL` and `n CHECKSEQUENCEVERIFY`
+    /// Ideally this should return int value as n: build_scriptint(t as i64)),
+    /// But since this is a T value and it will never be popped, it is safe to
+    /// push Satisfied.
+    fn evaluate_after<'desc, Pk, Pkh>(
+        &mut self,
+        n: &'desc u32,
+        age: &'desc u32,
+    ) -> Option<Result<SatisfiedConstraint<'desc, 'stack, Pk, Pkh>, Error>>
+        where Pk: ToPublicKey, Pkh: ToPublicKeyHash
+    {
+        if *age >= *n {
+            self.push(StackElement::Satisfied);
+            Some(Ok(SatisfiedConstraint::RelativeTimeLock { time: n }))
+        } else {
+            Some(Err(Error::RelativeLocktimeNotMet(n.clone())))
+        }
+    }
+
+    /// Helper function to evaluate a Older and OlderF Node. Takes no arugment from stack
+    /// `n CHECKLOCKTIMEVERIFY 0NOTEQUAL` and `n CHECKLOCKTIMEVERIFY`
+    /// Ideally this should return int value as n: build_scriptint(t as i64)),
+    /// But since this is a T value and it will never be popped, it is safe to
```

Comment:
It's not necessarily a T. The reason we don't need to copy the Script semantics is that Miniscript never evaluates integers.

Reply:
Yeah, forgot to fix this comment. This comment was based on the previous version on miniscript for timeT